modules/account-lib/src/coin/dot/transactionBuilder.ts

```diff
@@ -1,57 +1,506 @@
+   */
+  durationConfig({ maxDuration }: { maxDuration: number | undefined }): this {
+    if (maxDuration !== undefined) {
+      this.validateValue(new BigNumber(maxDuration));
+      this._eraPeriod = maxDuration;
+    }
+    return this;
+  }
+  /**
+   *
+   * The nonce for this transaction.
+   *
+   * @param {number} nonce
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  sequenceId(nonce: sequenceId): this {
+    const value = new BigNumber(nonce.value);
+    this.validateValue(value);
+    this._nonce = value.toNumber();
+    return this;
+  }
+  /**
+   *
+   * The tip to increase transaction priority.
+   *
+   * @param {number | undefined} tip optional
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  tip(tip: number | undefined): this {
+    if (tip !== undefined) {
+      this.validateValue(new BigNumber(tip));
+    }
+    this._tip = tip;
+    return this;
+  }
+
+  /**
+   *
+   * The number of the checkpoint block after which the transaction is valid
+   *
+   * @param {number} blockNumber
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  validity({ firstValid }: Validity): this {
+    this.validateValue(new BigNumber(firstValid));
+    this._blockNumber = firstValid;
+    return this;
+  }
+  /**
+   *
+   * The hash of the checkpoint block.
+   *
+   * @param {number} blockHash
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   * @see https://wiki.polkadot.network/docs/build-protocol-info#transaction-mortality
+   */
+  blockHash(blockHash: string): this {
+    this._blockHash = blockHash;
+    return this;
+  }
+  /**
+   *
+   * The current version for transaction format.
+   *
+   * @param {number} transactionVersion
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  transactionVersion(transactionVersion: number): this {
+    this._transactionVersion = transactionVersion;
+    return this;
+  }
+
+  /**
+   *
+   * The spec name for the registry.
+   *
+   * @param {specNameType} specName
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  specName(specName: specNameType): this {
+    this._specName = specName;
+    return this;
+  }
+
+  /**
+   *
+   * The genesis hash of the chain.
+   *
+   * @param {string} genesisHash
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  genesisHash(genesisHash: string): this {
```

Comment:
Doesn't this go to `statics`?

Reply:
needs dynamic setters as well. Stored in and called from statics when we use helper functions to setup the network

Reply:
Why does it need to be dynamic, the genesis hash is always the same for each network we setup, aka static configuration

Reply:
based on @rliberal-simtlix's request but okay we can make them fully static